pkg/cmd/kafka/consumergroup/resetoffset/reset_offset.go

```diff
@@ -0,0 +1,288 @@
+					return err
+				}
+			}
+
+			if opts.value == "" && (opts.offset == consumergroup.AbsoluteOffset || opts.offset == consumergroup.TimestampOffset) {
+				return errors.New(opts.localizer.MustLocalize("kafka.consumerGroup.resetOffset.error.valueRequired", localize.NewEntry("Offset", opts.offset)))
+			}
+
+			if opts.kafkaID != "" {
+				return runCmd(opts)
+			}
+
+			cfg, err := opts.Config.Load()
+			if err != nil {
+				return err
+			}
+
+			if !cfg.HasKafka() {
+				return errors.New(opts.localizer.MustLocalize("kafka.consumerGroup.common.error.noKafkaSelected"))
+			}
+
+			opts.kafkaID = cfg.Services.Kafka.ClusterID
+
+			return runCmd(opts)
+		},
+	}
+
+	cmd.Flags().BoolVarP(&opts.skipConfirm, "yes", "y", false, opts.localizer.MustLocalize("kafka.consumerGroup.resetOffset.flag.yes"))
+	cmd.Flags().StringVar(&opts.id, "id", "", opts.localizer.MustLocalize("kafka.consumerGroup.common.flag.id.description", localize.NewEntry("Action", "reset-offset")))
+	cmd.Flags().StringVar(&opts.value, "value", "", opts.localizer.MustLocalize("kafka.consumerGroup.resetOffset.flag.value"))
+	cmd.Flags().StringVar(&opts.offset, "offset", "", opts.localizer.MustLocalize("kafka.consumerGroup.resetOffset.flag.offset"))
+	cmd.Flags().StringVar(&opts.topic, "topic", "", opts.localizer.MustLocalize("kafka.consumerGroup.resetOffset.flag.topic"))
+	cmd.Flags().Int32SliceVar(&opts.partitions, "partitions", []int32{}, opts.localizer.MustLocalize("kafka.consumerGroup.resetOffset.flag.partitions"))
+	cmd.Flags().StringVarP(&opts.output, "output", "o", "", opts.localizer.MustLocalize("kafka.consumerGroup.resetOffset.flag.output"))
+
+	_ = cmd.MarkFlagRequired("id")
+	_ = cmd.MarkFlagRequired("offset")
+	_ = cmd.MarkFlagRequired("topic")
+
+	// flag based completions for ID
+	_ = cmd.RegisterFlagCompletionFunc("id", func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+		return cmdutil.FilterValidConsumerGroupIDs(f, toComplete)
+	})
+
+	// flag based completions for topic
+	_ = cmd.RegisterFlagCompletionFunc("topic", func(cmd *cobra.Command, _ []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+		return cmdutil.FilterValidTopicNameArgs(f, toComplete)
+	})
+
+	flagutil.EnableOutputFlagCompletion(cmd)
+	flagutil.EnableStaticFlagCompletion(cmd, "offset", flagutil.ValidOffsets)
+
+	return cmd
+}
+
+// nolint:funlen
+func runCmd(opts *Options) error {
+
+	conn, err := opts.Connection(connection.DefaultConfigRequireMasAuth)
+	if err != nil {
+		return err
+	}
+
+	api, kafkaInstance, err := conn.API().KafkaAdmin(opts.kafkaID)
+	if err != nil {
+		return err
+	}
+
+	ctx := context.Background()
+
+	offsetResetParams := kafkainstanceclient.ConsumerGroupResetOffsetParameters{
+		Offset: opts.offset,
+	}
+
+	if opts.value != "" {
+		offsetResetParams.Value = &opts.value
+	}
+
+	if opts.offset == consumergroup.AbsoluteOffset {
+		if _, parseErr := strconv.Atoi(opts.value); parseErr != nil {
```

Comment:
Should we have a generic method for this as well?

Reply:
I was thinking a generic method to check if string is a valid integer.

Reply:
Isn't that what `strconv.Atoi` does?

Reply:
Ignore last comment, implemented `Validator` Interface for consumergroups to make it consistent with topics.